manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/utils/CustomSnippetsUtil.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.generator.services.utils;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.commons.framework.patch.StrategicPatch;
+import io.hyscale.commons.io.HyscaleFilesUtil;
+import io.hyscale.commons.utils.DataFormatConverter;
+import io.hyscale.commons.utils.ObjectMapperFactory;
+import io.hyscale.servicespec.commons.builder.MapFieldDataProvider;
+import io.hyscale.servicespec.commons.fields.HyscaleSpecFields;
+import io.hyscale.servicespec.commons.model.service.ServiceSpec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Custom K8s Snippets
+ * <p>
+ * This class is responsible for patching custom K8s snippets on top of generated manifests.
+ * </p>
+ *
+ * @author Nishanth Panthangi
+ */
+
+@Component
+public class CustomSnippetsUtil {
+    private static final Logger logger = LoggerFactory.getLogger(CustomSnippetsUtil.class);
+
+    private static Map<String,List<String>> kindVsCustomSnippets = null;
+
+    public void initCustomSnippets(ServiceSpec serviceSpec) throws HyscaleException {
```

Comment:
If you want to have state use the bean scopes else
Don't have a state stored in the bean itself, because you will not be able to handle concurrency later.

Better extract the snippets and handle them at the caller side whenever required.